src/components/styled-components/StyledDiv.js

```diff
@@ -0,0 +1,24 @@
+import styled from 'styled-components';
+
+const StyledDiv = styled.div`
+	${({ variant }) =>
+		variant === 'dateinput' &&
+		`
+        display: flex;
+        align-items: center;
+    `}
```

Comment:
You repeat this part in the next variant. You could write that differently. Does this work properly at all? According to my understanding you make the variant query behind your css declaration, which should be variable. But it can also be that I just do not know how you do it.